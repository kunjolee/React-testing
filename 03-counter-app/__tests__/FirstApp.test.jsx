import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Tests on <FirstApp />', () => {
    it('should match the snapshot', () => {
        const { container } = render(<FirstApp title='Hi, im kunjo' />);
        expect(container).toMatchSnapshot();
    });

    it('should have the title in h1', () => {
        const title = 'Hi, im kunjo';
        const { container, getByText, getByTestId } = render(
            <FirstApp title={title} />
        );
        expect(getByText(title)).toBeTruthy();
        // const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toBe(title);
        expect(getByTestId('test-title').innerHTML).toContain(title);
    });

    it('should show the subtitle sent by props', () => {
        const title = 'Hi, im kunjo';
        const subTitle = 'Im a subtitle';
        const { getAllByText } = render(
            <FirstApp title={title} subTitle={subTitle} />
        );
        expect(getAllByText(subTitle).length).toBe(2);
    });
});
