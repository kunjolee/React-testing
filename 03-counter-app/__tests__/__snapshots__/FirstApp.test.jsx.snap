// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <FirstApp /> should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hi, im kunjo
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
